lib/ship_fosdick/individual_shipment_updater.rb

```diff
@@ -0,0 +1,68 @@
+module ShipFosdick
+  class IndividualShipmentUpdater
+    class InvalidTransitionError < StandardError; end
+
+    def initialize(manifest_row,
+                   shipment_repository: Spree::Shipment,
+                   invalid_transition_error_factory: InvalidTransitionError)
+
+      @manifest_row = ManifestRow.new(manifest_row)
+      @shipment_repository = shipment_repository
+      @invalid_transition_error_factory = invalid_transition_error_factory
+    end
+
+    def perform
+      return unless can_update?
+
+      raise_invalid_transition_error unless shipment.can_ship?
+
+      update_attributes
+      transition_state
+      update_order
+    end
+
+    alias_method :update, :perform
+
+    private
+
+    attr_reader :invalid_transition_error_factory, :manifest_row,
+      :shipment_repository
+
+    def can_update?
+      shipment &&
+        !shipment.shipped?
+    end
+
+    def raise_invalid_transition_error
+      raise invalid_transition_error_factory,
+        "Cannot transition shipment from current state: '#{shipment.state}' to requested state: 'shipped', no transition found."
+    end
+
+    def shipment
+      shipment_repository.find_by(number: manifest_row.shipment_number)
+    end
+
+    def transition_state
+      shipment.ship
+    end
+
+    def update_attributes
+      shipment.update(tracking: manifest_row.tracking_code)
+    end
+
+    def update_order
+      shipment.order.updater.update_shipment_state
```

Comment:
Can this be replaced with a simple [`order.update!`][1] call? It calls `updater.update`, and [`updater.update`][2] calls `update_shipment_state`.

[1]: https://github.com/spree/spree/blob/2-4-stable/core/app/models/spree/order.rb#L249-L251
[2]: https://github.com/spree/spree/blob/2-4-stable/core/app/models/spree/order_updater.rb#L17-L26